RobotCode/src/main/java/frc/lib/core/util/FileUtils.java

```diff
@@ -0,0 +1,85 @@
+package frc.lib.core.util;
+
+import com.google.gson.Gson;
+
+import java.io.*;
+import java.util.Scanner;
+
+public class FileUtils {
+
+	private static final Gson Gson = new Gson();
+	private static String toJson(Object object) {
+		return Gson.toJson(object);
+	}
+
+	private static <T> T fromJson(String json, Class<T> dataType) {
+		// I'm not sure (Class<T>) Object.class what is, but I'm trusting it.
+		return Gson.fromJson(json, dataType);
+	}
+
+	/**
+	 * Reads a json file and returns it as an object.
+	 * Don't do circular references.
+	 * <p>
+	 * Ex: If you do: class A { A b; } A a = new A(); a.b = a; You will get a StackOverflowError.
+	 * @param filePath The file path to read from.
+	 * @return The JsonObject returned from the file.
+	 * @throws FileNotFoundException Throws an error if the file does not exist.
+	 */
+	public static <T> T readObjectFromFilePath(String filePath, Class<T> classType) throws FileNotFoundException {
+		return readObjectFromFile(new File(filePath), classType);
+	}
+
+	/**
+	 * Reads a json file and returns it as an object.
+	 * Don't do circular references.
+	 * <p>
+	 * Ex: If you do: class A { A b; } A a = new A(); a.b = a; You will get a StackOverflowError.
+	 * @param file The file to read from.
+	 * @return The Object returned from the file.
+	 * @throws FileNotFoundException Throws an error if the file does not exist.
+	 */
+	public static <T> T readObjectFromFile(File file, Class<T> classType) throws FileNotFoundException {
```

Comment:
Make this private

Reply:
ok so the branch doesn't build when i make readObjectFromFile private, because the test extends from T a variable in the method, and i'm not smart enough to figure out why that is